app/migrations/Version20211020114811.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Schema\Schema;
+use Mautic\CoreBundle\Doctrine\AbstractMauticMigration;
+
+final class Version20211020114811 extends AbstractMauticMigration
+{
+    public function up(Schema $schema): void
+    {
+        $this->addSql('SET FOREIGN_KEY_CHECKS=0;');
+
+        /*
+         * The index key prefix length limit is 3072 bytes for InnoDB tables.
+         * In utf8mb4, 1 char uses 4 bytes
+         * So if we are creating multiple columns index for 4 columns of varchar(255) then it will take 255*4*4 = 4080.
+         * 4080 bytes is more than max allowed limit 3072 bytes.
+         * So it was not allowing to convert charset for particular tables and was showing error
+         * "Specified key was too long; max key length is 3072 bytes"
+         */
+
+        $dropIndexQuery = 'DROP INDEX %s on %s';
+
+        $this->addSql(
+            sprintf(
+                $dropIndexQuery,
+                $this->prefix.'company_match',
+                $this->getPrefixedTableName('companies')
+            )
+        );
```

Comment:
@dadarya0 this migration can fail with `Can't DROP INDEX `company_match`; check that it exists (uncaught exception)`. Why there aren't any `preUp` checks?

Reply:
I think preUp was not added for saving time in execution of migration otherwise it was taking too much time.

Reply:
We'll, we cannot drop index that doesn't exist. The check for the existence of the index is pretty much instant, so I'm not sure why it would be the case.

Reply:
Fixed in https://github.com/mautic/mautic/pull/14852. @dadarya0 please take a look if it's OK.